api/src/services/users/get_roles_and_privileges.py

```diff
@@ -0,0 +1,47 @@
+import logging
+from uuid import UUID
+
+from sqlalchemy import select
+from sqlalchemy.orm import selectinload
+
+from src.adapters import db
+from src.db.models.user_models import AgencyUser, ApplicationUser, OrganizationUser, User
+
+logger = logging.getLogger(__name__)
+
+
+def get_roles_and_privileges(
+    db_session: db.Session,
+    user_id: UUID,
+) -> User | None:
```

Comment:
If this is null, we should raise a 404 just so this function can always say it returns something. Now, in the context of this implementation, we can't actually hit this scenario (we know the user ID matches the user we authed, so that user MUST exist or auth would have failed), but should add that just to follow the convention. If we use this function for a different endpoint where that isn't true, it'd be already setup.